sim/midas/src/main/scala/midas/Config.scala

```diff
@@ -212,3 +222,81 @@ class VitisConfig extends Config(new Config((site, here, up) => {
 class HostDebugFeatures extends Config((site, here, up) => {
   case GenerateTokenIrrevocabilityAssertions => true
 })
+
+class WithWiringTransform extends Config((site, here, up) => {
+  case TargetTransforms => Dependency[firrtl.passes.wiring.WiringTransform] +: up(TargetTransforms, site)
+})
+
+class WithPCIMPorts extends Config((site, here, up) => {
+  case F1ShimHasPCIMPorts => true
+  case FPGAStreamEngineInstantiatorKey =>
+    (e: StreamEngineParameters, p: Parameters) =>
+    new FPGAManagedStreamEngine(p, e)
+  case FPGAManagedAXI4Key   => Some(FPGAManagedAXI4Params(
+    size                = BigInt(1) << 64,  // 128GB (size of Bar 4)
+    dataBits            = 512,                  // 512 bits set by FireSim default
+    idBits              = 6,                    // 6 is a guess; based upon CPUManagedAXI4Params
+    writeTransferSizes  = TransferSizes(512/8), // Default for Firesim
+    readTransferSizes   = TransferSizes(512/8), // Default for Firesim
+    interleavedId       = Some(0)               // Interleaved not currentlky supported therefore 0
+  ))
+})
+
+class EC2F1Config extends Config(
+  new WithPCIMPorts ++
+  new WithWiringTransform ++
+  new F1Config
+)
+
+class WithModelMultiThreading extends Config((site, here, up) => {
+  case midas.EnableModelMultiThreading => true
+})
+
+class WithMultiCycleRams extends Config((site, here, up) => {
+  case midas.GenerateMultiCycleRamModels => true
+})
+
+class MTModels extends WithModelMultiThreading
+class MCRams extends WithMultiCycleRams
+
+case object FireAxeNoCPartitionPass extends Field[Boolean](false)
+case object FireAxeQSFPConnections  extends Field[Boolean](false)
+case object FireAxePCIMConnections  extends Field[Boolean](false)
+case object FireAxePCISConnections  extends Field[Boolean](false)
+case object FireAxePreserveTarget   extends Field[Boolean](false)
+case object FireAxePartitionGlobalInfo extends Field[Option[Seq[Seq[String]]]](None)
+case object FireAxePartitionIndex extends Field[Option[Int]](None)
+
+class WithFireAxePreserveTarget extends Config((site, here, up) => {
+  case midas.FireAxePreserveTarget => true
+})
+
+class WithFireAxeNoCPart extends Config((site, here, up) => {
+  case midas.FireAxeNoCPartitionPass => true
+})
+
+class WithQSFP extends Config((site, here, up) => {
+  case midas.FireAxeQSFPConnections => true
+})
+
+class WithPCIS extends Config((site, here, up) => {
+  case midas.FireAxePCISConnections => true
+})
+
+class WithPCIM extends Config((site, here, up) => {
+  case midas.FireAxePCIMConnections => true
+})
+
+class WithPartitionGlobalInfo(info: Seq[Seq[String]]) extends Config((site, here, up) => {
+  case midas.FireAxePartitionGlobalInfo => Some(info)
+})
+
+class WithPartitionBase extends Config((site, here, up) => {
+  case midas.FireAxePartitionIndex => None
+})
```

Comment:
For these case classes can you disable the other partition modes if this is used (i.e. if someone accidently uses two different partition modes in the same config). Ex: Disable QSFP if PCIM used. vice versa since they should be mutually exclusive.